src/Nethermind/Nethermind.JsonRpc.TraceStore/TraceStoreRpcModule.cs

```diff
@@ -0,0 +1,311 @@
+            return ResultWrapper<IEnumerable<ParityTxTraceFromStore>>.Success(txTracerFilter.FilterTxTraces(txTraces));
+        }
+    }
+
+    public ResultWrapper<IEnumerable<ParityTxTraceFromStore>> trace_block(BlockParameter blockParameter)
+    {
+        SearchResult<BlockHeader> blockSearch = _blockFinder.SearchForHeader(blockParameter);
+        if (blockSearch.IsError)
+        {
+            return ResultWrapper<IEnumerable<ParityTxTraceFromStore>>.Fail(blockSearch);
+        }
+
+        BlockHeader block = blockSearch.Object!;
+        if (TryGetBlockTraces(block, out List<ParityLikeTxTrace>? traces) && traces is not null)
+        {
+            return ResultWrapper<IEnumerable<ParityTxTraceFromStore>>.Success(traces.SelectMany(ParityTxTraceFromStore.FromTxTrace));
+        }
+
+        return _traceModule.trace_block(blockParameter);
+    }
+
+    public ResultWrapper<IEnumerable<ParityTxTraceFromStore>> trace_get(Keccak txHash, long[] positions)
+    {
+        ResultWrapper<IEnumerable<ParityTxTraceFromStore>> traceTransaction = trace_transaction(txHash);
+        List<ParityTxTraceFromStore> traces = TraceRpcModule.ExtractPositionsFromTxTrace(positions, traceTransaction);
+        return ResultWrapper<IEnumerable<ParityTxTraceFromStore>>.Success(traces);
+    }
+
+    public ResultWrapper<IEnumerable<ParityTxTraceFromStore>> trace_transaction(Keccak txHash) =>
+        TryTraceTransaction(
+            txHash,
+            ParityTraceTypes.Trace,
+            static t => ParityTxTraceFromStore.FromTxTrace(t),
+            out ResultWrapper<IEnumerable<ParityTxTraceFromStore>>? result)
+        && result is not null
+            ? result
+            : _traceModule.trace_transaction(txHash);
+
+    private bool TryTraceTransaction<T>(
+        Keccak txHash,
+        ParityTraceTypes traceTypes,
+        Func<ParityLikeTxTrace, T> map,
+        out ResultWrapper<T>? result)
+    {
+        SearchResult<Keccak> blockHashSearch = _receiptFinder.SearchForReceiptBlockHash(txHash);
+        if (blockHashSearch.IsError)
+        {
+            {
+                result = ResultWrapper<T>.Fail(blockHashSearch);
+                return true;
+            }
+        }
+
+        SearchResult<Block> blockSearch = _blockFinder.SearchForBlock(new BlockParameter(blockHashSearch.Object!));
+        if (blockSearch.IsError)
+        {
+            {
+                result = ResultWrapper<T>.Fail(blockSearch);
+                return true;
+            }
+        }
+
+        Block block = blockSearch.Object!;
+
+        if (TryGetBlockTraces(block.Header, out List<ParityLikeTxTrace>? traces) && traces is not null)
+        {
+            ParityLikeTxTrace? trace = GetTxTrace(block, txHash, traces);
+            if (trace is not null)
+            {
+                FilterTrace(trace, traceTypes);
+                result = ResultWrapper<T>.Success(map(trace));
```

Comment:
NP: Another idea is to create a `Map` method to `ResultWrapper`.

Reply:
Not sure if I understood, but I think like this it is better separated to map just trace types instead of JSON RPC types